src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.vb

```diff
@@ -1556,6 +1557,94 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
+
+        Friend Overrides ReadOnly Property AllRequiredMembers As ImmutableSegmentedDictionary(Of String, Symbol)
+            Get
+                CalculateRequiredMembersIfRequired()
+                Debug.Assert(Not _lazyRequiredMembers.IsDefault)
+                Return _lazyRequiredMembers
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property HasRequiredMembersError As Boolean
+            Get
+                CalculateRequiredMembersIfRequired()
+                Debug.Assert(Not _lazyRequiredMembers.IsDefault)
+                Return _lazyRequiredMembers = s_requiredMembersErrorSentinel
+            End Get
+        End Property
+
+        Private Sub CalculateRequiredMembersIfRequired()
+            If Not _lazyRequiredMembers.IsDefault Then
+                Return
+            End If
+
+            Dim builder As ImmutableSegmentedDictionary(Of String, Symbol).Builder = Nothing
+            Dim success = TryCalculateRequiredMembers(builder)
+
+            Dim requiredMembers = If(success,
+                If(builder?.ToImmutable(), ImmutableSegmentedDictionary(Of String, Symbol).Empty),
+                s_requiredMembersErrorSentinel)
+
+            RoslynImmutableInterlocked.InterlockedInitialize(_lazyRequiredMembers, requiredMembers)
+        End Sub
+
+        Private Function TryCalculateRequiredMembers(ByRef requiredMembersBuilder As ImmutableSegmentedDictionary(Of String, Symbol).Builder) As Boolean
+            Dim lazyRequiredMembers = _lazyRequiredMembers
+            If lazyRequiredMembers = s_requiredMembersErrorSentinel Then
+                Return False
+            End If
+
+            If TryCast(BaseTypeNoUseSiteDiagnostics, PENamedTypeSymbol)?.TryCalculateRequiredMembers(requiredMembersBuilder) = False Then
+                Return False
+            End If
+
+            If (Not HasAnyDeclaredRequiredMembers) AndAlso requiredMembersBuilder Is Nothing Then
+                Return True
+            End If
+
+            ' Need to make sure that members from a base type weren't hidden by members from the current type. That is an error scenario
+
+            If requiredMembersBuilder Is Nothing Then
+                requiredMembersBuilder = ImmutableSegmentedDictionary.CreateBuilder(Of String, Symbol)()
+            End If
+
+            For Each member In GetMembersUnordered()
+                If requiredMembersBuilder.ContainsKey(member.Name) Then
+                    ' This is only permitted if the member is an override of a required member from a base type, and is required itself
+                    Dim overriddenMember = TryCast(member, PropertySymbol)?.OverriddenProperty
+
+                    If (Not member.IsRequired()) OrElse
+                       member.Kind = SymbolKind.Field OrElse
+                       overriddenMember Is Nothing OrElse
+                       (Not overriddenMember.Equals(requiredMembersBuilder(member.Name), TypeCompareKind.ConsiderEverything)) Then
```

Comment:
>ConsiderEverything [](http://example.com/codeflow?start=105&length=18)

Why ```ConsiderEverything```?

Reply:
Thinking about this a bit more, I actually don't think that I even need symbol comparison. The purpose of this check is "am I literally overriding this member", and reference equality should suffice for this.

Reply:
>  I literally overriding this member", and reference equality should suffice for this.

That is not a guarantee for generic scenarios for example. We might have equal, but distinct members from constructed base.

Reply:
BTW, since we are comparing for the purpose of overriding, we probably should pay attention to custom modifiers